lib/binTree/parsingTree.c

```diff
@@ -0,0 +1,171 @@
+#include "parsingTree.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+struct Node
+{
+    char* value;
+    Node* leftChild;
+    Node* rightChild;
+};
+
+Node* buildTree(char expression[100][100], int* pointer)
+{
+    char* item = expression[*pointer];
+    ++(*pointer);
+    if (!item[0])
+    {
+        return NULL;
+    }
+    if (strstr("+-/*", item))
+    {
+        Node* operator = calloc(1, sizeof(Node));
+        if (operator == NULL)
+        {
+            return NULL;
+        }
+        operator->value = calloc(2, sizeof(char));
+        if (operator->value == NULL)
+        {
+            return NULL;
+        }
+        strcpy(operator->value, item);
+
+        Node* operand1 = calloc(1, sizeof(Node*));
+        if (operand1 == NULL)
+        {
+            return NULL;
+        }
+        operand1 = buildTree(expression, pointer);
+        operator->leftChild = operand1;
+
+        Node* operand2 = calloc(1, sizeof(Node*));
+        if (operand2 == NULL)
+        {
+            return NULL;
+        }
+        operand2 = buildTree(expression, pointer);
+        operator->rightChild = operand2;
+
+        return operator;
+    }
+    else
+    {
+        Node* operand = calloc(1, sizeof(Node));
+        if (operand == NULL)
+        {
+            return NULL;
+        }
+        operand->value = calloc(strlen(item), sizeof(char));
+        strcpy(operand->value, item);
+        return operand;
+    }
+}
+
+Node* buildTreeFromFile(const char* filename)
+{
+    FILE* file;
```

Comment:
```suggestion
    FILE* file = NULL;
```
Стайлгайд запрещает неинициализированные переменные